platform/x11/cog-platform-x11.c

```diff
@@ -380,15 +390,7 @@ xcb_process_events (void)
             }
 
             if (client_message->type == XCB_ATOM_NOTICE) {
-                if (s_window->xcb.needs_initial_paint) {
-                    xcb_initial_paint ();
-                    s_window->xcb.needs_initial_paint = false;
-                }
-
-                if (s_window->xcb.needs_frame_completion) {
-                    xcb_frame_completion ();
-                    s_window->xcb.needs_frame_completion = false;
-                }
+                xcb_handle_notice();
```

Comment:
It feels just slightly odd to me that there's an `xcb_handle_notice` rather than just putting the code here like the message handlers above. It's not so big that breaking it out helps readability and it's not going to get reused anywhere else(?)

Reply:
I'm not 100% sure because I wrote this a while ago, but probably I thought the function would end up being longer, but then it didn't. Let's move back the code inline into the `switch`.